goinp/goinp_test.go

```diff
@@ -425,3 +428,227 @@ func TestSelectFromStringsFromReader(t *testing.T) {
 		require.EqualError(t, err, "invalid option: You entered a number greater than the last option's number")
 	}
 }
+
+func testAskOptionWrapper(title string, defaultValue string, optional bool, stdin string, options ...string) (string, string, error) {
+	in, err := ioutil.TempFile("/tmp", "gotest")
+	if err != nil {
+		return "", "", err
+	}
+	defer in.Close()
+
+	out, err := ioutil.TempFile("/tmp", "gotest")
+	if err != nil {
+		return "", "", err
+	}
+	defer in.Close()
+
+	// test single input - not optional
+	if _, err := io.WriteString(in, stdin); err != nil {
+		return "", "", err
+	}
+
+	if _, err = in.Seek(0, os.SEEK_SET); err != nil {
+		return "", "", err
+	}
+
+	os.Stdin = in
+	os.Stdout = out
+
+	var s string
+	if len(options) > 0 {
```

Comment:
Can just `s, err = AskOptions(title, defaultValue, optional, options...)` be called all the time?